File: MusicPlayerService/app/src/main/java/com/xyz/musicplayerservice/MainActivity.java
package com.xyz.musicplayerservice;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private Button mBtnStartMusic;
    private Button mBtnStopMusic;
    private Intent intent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initViewsAndListeners();
    }

    private void initViewsAndListeners() {
        mBtnStartMusic = findViewById(R.id.btnStartMusic);
        mBtnStopMusic = findViewById(R.id.btnStopMusic);
        mBtnStartMusic.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                intent = new Intent(MainActivity.this, MusicService.class);
                startService(intent);
                MusicService.enqueueWork(MainActivity.this, MusicService.class, 1, intent);
            }
        });

        mBtnStopMusic.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

            }
        });
    }
}

File: BasketBallScore_Viewmodel/app/src/main/java/com/xyz/basketballscore/MainActivity.kt
package com.xyz.basketballscore
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import kotlinx.android.synthetic.main.activity_main.*
class MainActivity : AppCompatActivity(), View.OnClickListener {
    private lateinit var scoreViewModel: ScoreViewModel
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initViewsAndListeners()
        scoreViewModel = ViewModelProvider(this).get(ScoreViewModel::class.java)
        getInitialScoresAndUpdate()
    }
    private fun getInitialScoresAndUpdate() {
        tvPhoenixScore.text = scoreViewModel.getPhoenixScore().toString()
        tvTrojansScore.text = scoreViewModel.getTrojansScore().toString()
    }
    private fun initViewsAndListeners() {
        btnTrojansThreePoints.setOnClickListener(this)
        btnTrojansTwoPoints.setOnClickListener(this)
        btnPhoenixThreePoints.setOnClickListener(this)
        btnPhoenixTwoPoints.setOnClickListener(this)
        btnResetScore.setOnClickListener(this)
    }
    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.btnTrojansThreePoints -> {
                scoreViewModel.updateTrojansScore(3)
                tvTrojansScore.text = scoreViewModel.getTrojansScore().toString()
            }
            R.id.btnTrojansTwoPoints -> {
                scoreViewModel.updateTrojansScore(2)
                tvTrojansScore.text = scoreViewModel.getTrojansScore().toString()
            }
            R.id.btnPhoenixThreePoints -> {
                scoreViewModel.updatePhoenixScore(3)
                tvPhoenixScore.text = scoreViewModel.getPhoenixScore().toString()
            }
            R.id.btnPhoenixTwoPoints -> {
                scoreViewModel.updatePhoenixScore(2)
                tvPhoenixScore.text = scoreViewModel.getPhoenixScore().toString()
            }
            R.id.btnResetScore -> {
                val resetScore = 0
                tvTrojansScore.text = resetScore.toString()
                tvPhoenixScore.text = resetScore.toString()
            }
        }
    }
}